src/main/java/org/openstreetmap/atlas/checks/commands/MapRouletteUploadCommand.java

```diff
@@ -0,0 +1,197 @@
+                final JsonDeserializationContext context) throws JsonParseException
+        {
+            final JsonObject full = json.getAsJsonObject();
+            final JsonObject properties = full.get(PROPERTIES).getAsJsonObject();
+            final String challengeName = properties.get(GENERATOR).getAsString();
+            final String instruction = properties.get(INSTRUCTIONS).getAsString();
+            final String taskID = properties.get(ID).getAsString();
+            final JsonArray geojson = full.get(FEATURES).getAsJsonArray();
+            final Task result = new Task();
+            result.setChallengeName(challengeName);
+            // Note: since the points are contained inside the geojson feature, and we're
+            // re-serializing on our way to MapRoulette, we don't need to explicitly parse out the
+            // points from the geojson. If you're considering pulling this class out for more
+            // general use, points should be parsed as well.
+            result.setGeoJson(Optional.of(geojson));
+            result.setInstruction(instruction);
+            result.setProjectName(projectName);
+            result.setTaskIdentifier(taskID);
+            return result;
+        }
+    }
+
+    private static final Switch<File> INPUT_DIRECTORY = new Command.Switch<>("logfiles",
+            "Path to folder containing log files to upload to MapRoulette.", File::new, Optionality.REQUIRED);
+    private static final Switch<File> CONFIG_LOCATION = new Command.Switch<>("config",
+            "Path to a file containing MapRoulette challenge configuration.", File::new, Optionality.REQUIRED);
+    private static final String PARAMETER_CHALLENGE = "challenge";
+    private static final Logger logger = LoggerFactory.getLogger(MapRouletteUploadCommand.class);
+    private final Map<String, Challenge> checkNameChallengeMap;
+
+    public MapRouletteUploadCommand()
+    {
+        super();
+        this.checkNameChallengeMap = new HashMap<>();
+    }
+
+    @Override protected void execute(final CommandMap commandMap, final MapRouletteConfiguration configuration)
+    {
+        final File[] files = ((File) commandMap.get(INPUT_DIRECTORY)).listFiles();
+        if (files == null)
+        {
+            logger.error("No files found for the given directory.");
+        }
+        else
+        {
+            try
+            {
+                final Gson gson = new GsonBuilder().registerTypeAdapter(Task.class, new TaskDeserializer(configuration.getProjectName())).create();
+                final Configuration instructions = this.loadConfiguration(commandMap);
+                for (final File logFile : files)
+                {
+                    try (BufferedReader reader = new BufferedReader(new FileReader(logFile.getPath())))
+                    {
+                        reader.lines().forEach(line -> {
+                            final Task task = gson.fromJson(line, Task.class);
```

Comment:
What happens if a non-log file is encountered?

Reply:
Good point. Should be handled now, thanks!